rpc/types/src/json.rs

```diff
@@ -86,6 +95,568 @@ define_request_and_response! {
     }
 }
 
+define_request_and_response! {
+    get_block_count,
+    cc73fe71162d564ffda8e549b79a350bca53c454 =>
+    core_rpc_server_commands_defs.h => 919..=933,
+    GetBlockCount,
+
+    // There are no request fields specified,
+    // this will cause the macro to generate a
+    // type alias to `()` instead of a `struct`.
+    Request {},
+
+    #[derive(Copy)]
+    ResponseBase {
+        count: u64,
+    }
+}
+
+define_request_and_response! {
+    on_get_block_hash,
+    cc73fe71162d564ffda8e549b79a350bca53c454 =>
+    core_rpc_server_commands_defs.h => 935..=939,
+    OnGetBlockHash,
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = OnGetBlockHashRequest { block_height: [3] };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, "[3]");
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    #[derive(Copy)]
+    Request {
+        // This is `std::vector<u64>` in `monerod` but
+        // it must be a 1 length array or else it will error.
+        block_height: [u64; 1],
+    },
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = OnGetBlockHashResponse { block_hash: String::from("asdf") };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, "\"asdf\"");
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    Response {
+        block_hash: String,
+    }
+}
+
+define_request_and_response! {
+    submit_block,
+    cc73fe71162d564ffda8e549b79a350bca53c454 =>
+    core_rpc_server_commands_defs.h => 1114..=1128,
+    SubmitBlock,
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = SubmitBlockRequest { block_id: String::from("asdf") };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, "\"asdf\"");
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    Request {
+        block_id: String,
+    },
+    /// ```rust
+    /// use serde_json::*;
+    /// use cuprate_rpc_types::json::*;
+    ///
+    /// let x = SubmitBlockResponse { status: String::from("asdf") };
+    /// let x = to_string(&x).unwrap();
+    /// assert_eq!(x, "\"asdf\"");
+    /// ```
+    #[cfg_attr(feature = "serde", serde(transparent))]
+    #[repr(transparent)]
+    Response {
+        status: String,
+    }
+}
```

Comment:
Is this type correct?

```cpp
  struct COMMAND_RPC_SUBMITBLOCK
  {
    typedef std::vector<std::string> request;
    
    struct response_t: public rpc_response_base
    {
      std::string block_id;

      BEGIN_KV_SERIALIZE_MAP()
        KV_SERIALIZE_PARENT(rpc_response_base)
        KV_SERIALIZE(block_id)
      END_KV_SERIALIZE_MAP()
    };
    typedef epee::misc_utils::struct_init<response_t> response;
  };

```

Reply:
No, I think I got confused. [`block_id` is documented as `status`](https://www.getmonero.org/resources/developer-guides/daemon-rpc.html#submit_block), are the docs wrong?

Reply:
`status` is in `rpc_response_base`,  `block_id` is a seperate field that was added here: https://github.com/monero-project/monero/commit/e8cac61f4b9a662cbc1b00e46d1f9a3dd991c5f0 I guess the docs were not updated so yes.

Reply:
OK, the output would be something like this, right?:
```json
{
  "result": {
    "status": "OK",
    "untrusted": false,
    "block_id": "<hash>"
  },
  "id": "0",
  "jsonrpc": "2.0"
}
```

Reply:
I think so yes